ckanext/datastore/blueprint.py

```diff
@@ -64,30 +70,109 @@ def dump(resource_id):
             )
         )
 
-    response = make_response()
-    response.headers[u'content-type'] = u'application/octet-stream'
+    fmt = data[u'format']
+    offset = data[u'offset']
+    limit = data.get(u'limit')
+    options = {u'bom': data[u'bom']}
+    sort = data[u'sort']
+    search_params = {
+        k: v
+        for k, v in data.items()
+        if k in [
+            u'filters', u'q', u'distinct', u'plain', u'language',
+            u'fields'
+        ]
+    }
 
-    try:
-        dump_to(
-            resource_id,
-            response,
-            fmt=data[u'format'],
-            offset=data[u'offset'],
-            limit=data.get(u'limit'),
-            options={u'bom': data[u'bom']},
-            sort=data[u'sort'],
-            search_params={
-                k: v
-                for k, v in data.items()
-                if k in [
-                    u'filters', u'q', u'distinct', u'plain', u'language',
-                    u'fields'
-                ]
-            },
+    if fmt == u'csv':
+        writer_factory = csv_writer
+        records_format = u'csv'
+        content_disposition = u'attachment; filename="{name}.csv"'.format(
+                                    name=encode_rfc2231(resource_id))
```

Comment:
it looks like this does nothing without language/charset set? https://docs.python.org/3/library/email.utils.html#email.utils.encode_rfc2231